Doc/library/bisect.rst

```diff
@@ -53,8 +56,11 @@ The following functions are provided:
    ``all(val > x for val in a[i : hi])`` for the right side.
 
    *key* specifies a :term:`key function` of one argument that is used to
-   extract a comparison key from each input element.  The default value is
-   ``None`` (compare the elements directly).
+   extract a comparison key from each element in the array.  To support
+   searching complex records, the key function is not applied to the *x* value.
```

Comment:
> To support searching complex records,

"searching complex records" seems a bit opaque to me (though it makes more sense in the context of the example). What about something like "To support searching for a particular comparison key"?

Reply:
This intent was to show a progression from the 1-D example in the first paragraph to the 2-D example in the second paragraph.   The 1-D "numeric table lookups" are the simple case and don't need the key-function. In contrast, the 2-D "complex records" require a key-function to isolate the comparison field (like an ORDER BY clause in SQL).

How about "list of tuples" instead of "complex records"?

> The :func:`bisect`function and :func:`insort` functions also work with lists of
> tuples.  The *key* argument serves to extract the field used for ordering
> records in a table::

Reply:
Either way sounds fine to me, and I think this PR is an overall good improvement any way you slice it.

My only minor concern is putting "tutorial"-like language in the "specification"-like parts of the doc entry. I would prefer to somehow make clear that the records need not be tuples and key need not be itemgetter/attrgetter. We could even remove the qualification of that part and just have the complete sentence
```
The key function is not applied to the *x* value.
```
Or maybe
```
To support searching by key as in [the example below](...),
the key function is not applied to the *x* value.
```

Reply:
I get that.  However, the motivation is the key to understanding and remembering the API.  Otherwise, we get the perpetual, "I was surprised ...".